### PR TITLE
enhance: Make channel meta able to writer 200k plus segments

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #33125
pr: #33279

### CODE REVIEW

File: internal/datacoord/channel_store.go [hunk 1]
```diff
@@ -146,7 +147,11 @@ func (op *ChannelOp) BuildKV() (map[string]string, []string, error) {
 		k := buildNodeChannelKey(op.NodeID, ch.GetName())
 		switch op.Type {
 		case Add, Watch, Release:
-			info, err := proto.Marshal(ch.GetWatchInfo())
+			tmpWatchInfo := ch.GetWatchInfo()
```
Comment (czs007) [inline internal/datacoord/channel_store.go:150]: no need to Clone?

Comment (XuanYang-cn) [reply internal/datacoord/channel_store.go:150]: Its very lite clone, don't want to update in place so that segmentIDs are set unexpectedly

**Pull request merged**